html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+
+### Learning Outcomes
+
+* When should you use a CSS transition?
+* When should you use a CSS animation?
+* What are the long and short hand notations for both?
+* What should you keep in mind regarding repaints?
+* What are the two distinct parts of creating a keyframe animation?
+
+### Transitions
+CSS transitions let you animate a change from an element's initial state to an end state. Think of an ordinary button element with a white background. When your mouse is away from the button, it's just sitting there. Boring. Then when you hover your mouse cursor over the button the background color smoothly transitions from white, to grey, to black, over a period of time. This is a CSS transition.
+
+When your mouse cursor was away from the button, the button was in the initial state. When you hovered over it you introduced the end state, the hover state, while the transition happened as the color smoothly faded from white to black.
+
+In the above example we animated the CSS `background-color` property. While there are plenty of CSS properties you can animate, you can't animate them all. So keep [this list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties) of "CSS animatable" properties handy so you can check whether or not you can do the transition you want.
+
+Now let's make a basic transition. There are two ways of writing a transition. One is the long form and the other is the short form. The long form looks like this:
+
+~~~css
+button {
+  /* ... other css properties ... */
+  background-color: white;
+  transition-property: background-color;
+  transition-duration: 2s;
+  transition-timing-function: ease-out;
+  transition-delay: 1s;
+}
+
+button:hover {
+  background-color: black;
+}
+~~~
+
+The `transition-property` tells us what CSS property will be animated. In this case it is the `background-color` which will change from white to black. The `transition-duration` means the fade from white to black will gradually happen over 2 seconds. The `transition-timing-function` lets us change the speed of the transition over the course of its duration. Here it will `ease-out`, meaning the color change will be faster at the start than at the end of the transition. The `transition-delay` means the transition will start 1 second after the cursor rests on the button. The short form of this is:
+
+~~~css
+button {
+  /* ... other css properties ... */
+  background-color: white;
+  transition: background-color 2s ease-out 1s;
+}
+
+button:hover {
+  background-color: black;
+}
+~~~
+
+The above transition will occur when the user hovers their mouse over the button as indicated by the `:hover` pseudo-class. Different pseudo-classes may be used such as `:active` or `:focus` and you can find the [full list here](https://developer.mozifont-sizella.org/en-US/docs/Web/CSS/Pseudo-classes). We will now explore each of those pseudo-classes one by one.
```

Comment:
no need to go into pseudo-classes, that's covered by another lesson.  I'd leave the first sentence and then say something like:

> In addition to :hover, or any of the other psuedo-classes  you can also trigger transitions by adding or removing classes with JavaScript. For example, clicking a button could append the `open` class to a dropdown menu, which would trigger the opening transition.